src/apps/media/src/app-revamp/components/Header.tsx

```diff
@@ -1,30 +1,138 @@
-import { Box, Typography, Button } from "@mui/material";
+import { MouseEvent, useState } from "react";
+import {
+  Box,
+  Typography,
+  Button,
+  IconButton,
+  Menu,
+  MenuItem,
+  ListItemText,
+  ListItemIcon,
+} from "@mui/material";
 import FileUploadIcon from "@mui/icons-material/FileUpload";
+import ArrowDropDownIcon from "@mui/icons-material/ArrowDropDown";
+import CreateNewFolderIcon from "@mui/icons-material/CreateNewFolder";
+import DriveFileRenameOutlineIcon from "@mui/icons-material/DriveFileRenameOutline";
+import DeleteIcon from "@mui/icons-material/Delete";
+import { RenameFolderDialog } from "./RenameFolderDialog";
+import { NewFolderDialog } from "./NewFolderDialog";
+import { DeleteFolderDialog } from "./DeleteFolderDialog";
 
 interface Props {
   title: string;
+  id?: string;
+  binId?: string;
+  groupId?: string;
 }
 
-export const Header = ({ title }: Props) => {
+export const Header = ({ title, id, binId, groupId }: Props) => {
+  const [openRenameDialog, setOpenRenameDialog] = useState(false);
+  const [openNewFolderDialog, setOpenNewFolderDialog] = useState(false);
+  const [openDeleteDialog, setOpenDeleteDialog] = useState(false);
```

Comment:
Since these dialogs are mutually exclusive, I'd prefer something like this stylistically:

```ts
type Dialogs = 'delete' | 'rename' | 'new' | null
const [openDialog, setOpenDialog] = useState<Dialogs>(null)
```

Wondering what your thoughts on this would be?